src/lib/core/fiber_channel.c

```diff
@@ -190,6 +193,23 @@ fiber_channel_check_wait(struct fiber_channel *ch, ev_tstamp start_time,
 
 void
 fiber_channel_close(struct fiber_channel *ch)
+{
+	if (!fiber_channel_graceful_close || ch->count == 0) {
+		/* Destroy if no graceful close or buffer is already empty. */
+		fiber_channel_destroy(ch);
+		return;
+	}
+	ch->is_write_closed = true;
```

Comment:
Here I'm in doubt: in the new behavior `close()` sets `is_write_closed` and doesn't set `is_closed`. It feels lopsided. I have no certain recommendations how to make it more clean, but would appreciate ideas and ready to discuss variants.

Reply:
Renamed `is_closed` -> `is_destroyed` and `is_write_closed` -> `is_closed`.

Reply:
It looks better, thanks!